src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp

```diff
@@ -2975,3 +2980,368 @@ void ConptyRoundtripTests::NewLinesAtBottomWithBackground()
+                 L"does. When it prints a wrapped line, we should make sure to "
+                 L"not break the line wrapping.");
+
+    // The importantly valuable variable here ended up being
+    // writingMethod=PrintWithWriteCharsLegacy. That was the one thing that
+    // actually repro'd this bug. The other variables were introduced as part of
+    // debugging, and are left for completeness.
+    BEGIN_TEST_METHOD_PROPERTIES()
+        TEST_METHOD_PROPERTY(L"Data:circledRows", L"{2, 4, 10}")
+        TEST_METHOD_PROPERTY(L"Data:paintEachNewline", L"{0, 1, 2}")
+        TEST_METHOD_PROPERTY(L"Data:writingMethod", L"{0, 1}")
+    END_TEST_METHOD_PROPERTIES();
+    constexpr int DontPaint = 0;
+    constexpr int PaintAfterBothLines = 1;
+    constexpr int PaintEveryLine = 2;
+
+    constexpr int PrintWithPrintString = 0;
+    constexpr int PrintWithWriteCharsLegacy = 1;
+
+    INIT_TEST_PROPERTY(int, writingMethod, L"Controls using either ProcessString or WriteCharsLegacy to write to the buffer");
+    INIT_TEST_PROPERTY(int, circledRows, L"Controls the number of lines we output.");
+    INIT_TEST_PROPERTY(int, paintEachNewline, L"Controls whether we should call PaintFrame every line of text or not.");
+
+    auto& g = ServiceLocator::LocateGlobals();
+    auto& renderer = *g.pRender;
+    auto& gci = g.getConsoleInformation();
+    auto& si = gci.GetActiveOutputBuffer();
+    auto& sm = si.GetStateMachine();
+    auto* hostTb = &si.GetTextBuffer();
+    auto* termTb = term->_buffer.get();
+
+    _flushFirstFrame();
+
+    _checkConptyOutput = false;
+    _logConpty = true;
+
+    const size_t width = static_cast<size_t>(TerminalViewWidth);
+
+    const auto charsInFirstLine = width;
+    const auto numCharsFirstColor = 30;
+    const auto numCharsSecondColor = charsInFirstLine - numCharsFirstColor;
+    const auto charsInSecondLine = width / 2;
+
+    const auto defaultAttrs = TextAttribute();
+    const auto conhostDefaultAttrs = si.GetAttributes();
+
+    // Helper to abstract calls into either StateMachine::ProcessString or
+    // WriteCharsLegacy
+    auto print = [&](const std::wstring_view str) {
+        if (writingMethod == PrintWithPrintString)
+        {
+            sm.ProcessString(str);
+        }
+        else if (writingMethod == PrintWithWriteCharsLegacy)
+        {
+            doWriteCharsLegacy(si, str, WC_LIMIT_BACKSPACE);
+        }
+    };
+
+    // Each of the lines of text in the buffer will look like the following:
+    //
+    // Line 1 chars: ~~~~~~~~~~~~~~~~~~ <wrap>
+    // Line 1 attrs: YYYYYYYDDDDDDDDDDD (First 30 are yellow FG, then default attrs)
+    // Line 2 chars: ~~~~~~~~~~________ <break> (there are width/2 '~'s here)
+    // Line 2 attrs: DDDDDDDDDDDDDDDDDD (all are default attrs)
+    //
+    // The last line of the buffer will be used as a "prompt" line, with a
+    // single ':' in it. This is similar to the way `less` typically displays
+    // it's prompt at the bottom of the buffer.

```

Comment:
I don't know what this looks like offhand. Can you give the sample of the prompt line too since that's the BFD here?